src/lib/coda_lib/coda_lib.ml

```diff
@@ -784,8 +789,16 @@ module Make (Inputs : Inputs_intf) = struct
               (Linear_pipe.iter (Snark_pool.broadcasts snark_pool) ~f:(fun x ->
                    Net.broadcast_snark_pool_diff net x ;
                    Deferred.unit )) ;
+            let proposer_kps_r, proposer_kps_w =
+              Strict_pipe.(
```

Comment:
Why can't this be a broadcast_pipe or an incremental variable? I think it would make the semantics of updating the proposer keys easier to read. Why does this have to be async?

Reply:
I wanted the proposer keys to live in the proposer and not (also) coda_lib. The nice thing about the pipes here is that it gets drained.

Reply:
@nholland94 . Oh, to be more specific, a synchronous pipe with a cache on top of it.